### PR TITLE
docs: Update OpenAPI for tracker exporter endpoints [DHIS2-16019]

### PR DESCRIPTION
**Pull request by enricocolasante**
Update OpenAPI for endpoints using new `StartDateTime` and `EndDateTime` parameter types.
Using the changes from [ in](https://github.com/dhis2/dhis2-core/pull/17631)

This is how the generated specs look like:
Screenshot from 2024-05-30 09-05-05

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/resources/openapi/EnrollmentsExportController.md [hunk 1]
```diff
@@ -26,11 +26,13 @@ for how to use it.
 
 ### `*.parameter.EnrollmentRequestParams.enrolledAfter`
 
-Get enrollments with an enrollment date after the given date.
+Get enrollments enrolled after the given date and time. If no time is specified,
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/resources/openapi/EnrollmentsExportController.md:29]: I think the descriptions miss that before/after are inclusive.

**Review by teleivo — COMMENTED** [review]
Can you please add descriptions for the types StartDateTime and EndDateTime as well. @jbee can tell you what the best way is as it might have changed in the latest refactoring.

**Comment by jbee** [comment]
> ... can tell you what the best way is as it might have changed in the latest refactoring.

I wish :joy: I would hope you could annotate the new classes with `@OpenApi.Description` but I am unsure if I implemented such a fallback. If not we maybe should so one can provide a text that is used when the type occurs and no parameter specific description is given.

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/EndDateTime.java [hunk 2]
```diff
@@ -41,6 +42,10 @@
  * <p>This behavior, combined with {@link StartDateTime}, allows to correctly implement an interval
  * search including start and end dates.
  */
+@OpenApi.Description(
+    "EndDateTime represents an upper limit date and time used to filter results in search APIs. "
+        + "EndDateTime accepts any date and time in ISO8601 format. "
+        + "If no time is defined, then the time at the end of the day is used by default.")
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/EndDateTime.java:48]:
```suggestion
        + "The time defaults to the end of the day if no time is given.")
```

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/StartDateTime.java [hunk 3]
```diff
@@ -41,6 +42,10 @@
  * <p>This behavior, combined with {@link EndDateTime}, allows to correctly implement an interval
  * search including start and end dates.
  */
+@OpenApi.Description(
+    "StartDateTime represents a lower limit date and time used to filter results in search APIs. "
+        + "StartDateTime accepts any date and time in ISO8601 format. "
+        + "If no time is defined, then the time at the beginning of the day is used by default.")
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/StartDateTime.java:48]:
```suggestion
        + ""The time defaults to the beginning of the day if no time is given."")
```

File: dhis-2/dhis-web-api/src/main/resources/openapi/EnrollmentsExportController.md [hunk 4]
```diff
@@ -26,11 +26,15 @@ for how to use it.
 
 ### `*.parameter.EnrollmentRequestParams.enrolledAfter`
 
-Get enrollments with an enrollment date after the given date.
+Get enrollments enrolled after the given date and time.
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/resources/openapi/EnrollmentsExportController.md:29]: I would put all into one paragraph without newlines.

Does the ISO8601 format show up in the final spec? If not we need to describe that here as well as users should now what format we expect.

**Review by teleivo — APPROVED** [review]

**Pull request merged**